rm-datasource/src/main/java/io/seata/rm/datasource/AbstractPreparedStatementProxy.java

```diff
@@ -49,14 +51,10 @@
     /**
      * The Parameters.
      */
-    protected ArrayList<Object>[] parameters;
+    protected Map<Integer,ArrayList<Object>> parameters;
```

Comment:
I think we should use a dynamic array instead of HashMap, because only random access capability is needed.